src/molara/structure/io/importer_crystal.py

```diff
@@ -97,13 +100,7 @@ def load(self, use_pymatgen: bool = True) -> Crystals:
         """
         if use_pymatgen:
             try:
-                from monty.io import zopen
-                from pymatgen.core import Structure as PymatgenStructure
-
-                with zopen(self.path, "rt", errors="replace") as f:
-                    contents = f.read()
-                structure = PymatgenStructure.from_str(contents, fmt="poscar")
-                crystal = Crystal.from_pymatgen(structure, supercell_dims=[1, 1, 1])
+                crystal = PymatgenImporter(self.path).load().get_current_mol()
```

Comment:
The idea to use this routine rather than `PymatgenImporter` directly was that `pymatgen` has it's own patterns to recognize the file type (more specifically, the file name must contain `poscar` or `contcar` or `.vasp`). We would have to use the same patterns if we want to hand the `self.path` to `PymatgenImporter` and _not_ raise some error then in `pymatgen`. That's why I copied the internal `pymatgen` routine over to here. Would it be ok to keep it or do you prefer to change it?

Reply:
Essentially, I just want to keep the code as concise as possible. Could we maybe use the solution with monty in both cases? @ab5424

Reply:
I have changed the implementation of PymatgenImporter to use the `from_str` routine. I would like to keep that PoscarImporter doesn't have its own way with pymatgen but uses the implementation in PymatgenImporter. I think the code is cleaner like that. Are you fine with it, @ab5424?